actions/global_actions.jsx

```diff
@@ -474,12 +474,19 @@ export function emitSearchMentionsEvent(user) {
     if (user.notify_props) {
         const termKeys = UserStore.getMentionKeys(user.id);
 
-        if (termKeys.indexOf('@channel') !== -1) {
-            termKeys[termKeys.indexOf('@channel')] = '';
+        const indexOfChannel = termKeys.indexOf('@channel');
+        if (indexOfChannel !== -1) {
+            termKeys.splice(indexOfChannel, 1);
         }
 
-        if (termKeys.indexOf('@all') !== -1) {
-            termKeys[termKeys.indexOf('@all')] = '';
+        const indexOfAll = termKeys.indexOf('@all');
+        if (indexOfAll !== -1) {
+            termKeys.splice(indexOfAll, 1);
+        }
+
+        const indexOfHere = termKeys.indexOf('@here');
+        if (indexOfHere !== -1) {
+            termKeys.splice(indexOfHere, 1);
```

Comment:
This is the actual bug fix - suspect that we changed default mention keys to include `@here`, but didn't update this code. The rest of the change is just removing terms from the array, rather than replacing them with empty string to avoid building up unnecessary whitespace when the array elements are joined on line 492

Reply:
I've checked this issue and it might be related to this added key here - https://github.com/mattermost/mattermost-webapp/blob/master/stores/user_store.jsx#L464